FightPandemics/Features/Post/CreatePost/OfferRequest/OfferRequest.swift

```diff
@@ -0,0 +1,135 @@
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+
+import UIKit
+
+class OfferRequest: UIView {
+    private var topLine = UIView()
+    private var questionLbl = UILabel()
+    private var offerBtn = UIButton()
+    private var reqBtn = UIButton()
+    private var closeBtn = UIButton()
+    private var confirmBtn = UIButton()
+    private var firstVSV = UIStackView()
+    private var firstHSV = UIStackView()
+    private var finalVSV = UIStackView()
+    func setUp () {
+        setUpUpper()
+        setUpLower()
+        setUpConstraints()
+    }
+    func setUpUpper () {
+        topLine.backgroundColor = UIColor(hexString: "#F3F4FE")
+        topLine.frame.size.width = 116
+        topLine.frame.size.height = 4
+        topLine.clipsToBounds = true
+        topLine.layer.masksToBounds = true
+        topLine.layer.cornerRadius = topLine.frame.size.height / 2
+        self.addSubview(topLine)
+        questionLbl.text = "OfferRequestTitle".localized
+        questionLbl.font = Fonts.poppinsBold.customFont(size: 16)
+        questionLbl.textAlignment = .center
+        questionLbl.numberOfLines = 0
+        self.addSubview(questionLbl)
+    }
+    func setUpLower () {
+        offerBtn.setAttributedTitle(NSAttributedString(string: "OfferRequestOfferBtn".localized, attributes: [NSAttributedString.Key.font: Fonts.dmSansRegular.customFont(size: 12), NSAttributedString.Key.foregroundColor: UIColor.black]), for: .normal)
+        offerBtn.setTitleColor(UIColor.white, for: .selected)
+        offerBtn.frame.size.height = 45
+        offerBtn.frame.size.width = UIScreen.main.bounds.width - 46
+        offerBtn.backgroundColor = UIColor(hexString: "#F2F2F2")
+        offerBtn.clipsToBounds = true
+        offerBtn.layer.masksToBounds = true
+        offerBtn.layer.cornerRadius = 4
+        self.addSubview(offerBtn)
+        reqBtn.setAttributedTitle(NSAttributedString(string: "OfferRequestRequestBtn".localized, attributes: [NSAttributedString.Key.font: Fonts.dmSansRegular.customFont(size: 12), NSAttributedString.Key.foregroundColor: UIColor.black]), for: .normal)
+        reqBtn.setTitleColor(UIColor.white, for: .selected)
+        reqBtn.frame.size.height = 45
+        reqBtn.frame.size.width = UIScreen.main.bounds.width - 46
```

Comment:
Also on the buttons, I don't think we actually need to set the button frame height and width since we are using autolayout to handle that already.

Reply:
First, I needed the height for corner radius in cases of rounded buttons, and also there were cases where the button was not drawn without direct width and height. But I'll try it again and see how it goes